### PR TITLE
graphQL in frontend

### PR DESCRIPTION
**Pull request by trexrush**
setup apollo client to make graphql calls on the frontend

added a temporary route for requesting new user info with react-hook-forms

### CODE REVIEW

**Review by DedsecKnight — DISMISSED** [review]

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**/**](https://portal-next-qxlpgg91l-acmutd.vercel.app/) | [**/profile/update**](https://portal-next-qxlpgg91l-acmutd.vercel.app/profile/update) |
|:-:|:-:|
|<a href="https://portal-next-qxlpgg91l-acmutd.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/kKm4hTJb4pbjCBCWR1s2lsEl4PMODQpK-MVtvbxuvhvRF4K1Flf3cxDiPp7jhJvEZUBZ11AAyg19wNrtFgiuww.png">(view full size)</a> |<a href="https://portal-next-qxlpgg91l-acmutd.vercel.app/profile/update">Screenshot of /profile/update</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4Ow8UI2AZJzCWBgdLRkem562eVJ8bmLEvHwHoYTpMBZVku9206xu_Gy7kKHxUxzhe-n28o0RJ3AvtmpAGRR16g.png">(view full size)</a> |



And 1 other route:
- [**/api/auth/\[...nextauth\]**](https://portal-next-qxlpgg91l-acmutd.vercel.app/api/auth/[...nextauth])



Commit 3439999fad650a05fad9547b8e3757e667c5b729 (https://portal-next-qxlpgg91l-acmutd.vercel.app).

File: pages/index.tsx [hunk 1]
```diff
@@ -1,10 +1,48 @@
 import { signOut, useSession } from 'next-auth/react';
 import Link from 'next/link';
 
+import { useQuery, gql } from '@apollo/client';
+import { useEffect } from 'react';
+import { useRouter } from 'next/router';
+
+// const TEST = gql`
```
Comment (stevenmqnguyen) [inline pages/index.tsx]: Remove

File: pages/profile/update.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        major: result.major,
+        netid: result.netid,
+        user: session.id,
+        utdStudent: result.utdStudent,
+      },
+    },
+  });
+
+  useEffect(() => {
+    console.log(result);
```
Comment (stevenmqnguyen) [inline pages/profile/update.tsx]: remove unnecessary console.logs

**Review by stevenmqnguyen — DISMISSED** [review]

**Comment by stevenmqnguyen** [comment]
Looks great! Robust and clean. We'll have to use dataloader or something similar to solve N+1 issue from hasProfile resolver (and friends).

**Review by stevenmqnguyen — APPROVED** [review]

**Pull request merged**